### PR TITLE
Add auto refresh on/off config and session life time config.

### PR DESCRIPTION
**Pull request by ciremusyoka**
Adds functionality to only refresh token when:
- A config (EXPRESS_ALLOW_TOKEN_RENEWAL) is set to true
- Session stays active time(specified number of seconds) is not exceeded. Controlled byEXPRESS_MAXIMUM_SESSION_LIFE_TIME env

Addresses https://github.com/onaio/reveal-frontend/issues/1477

### CODE REVIEW

File: src/app/index.ts [hunk 1]
```diff
@@ -160,12 +168,22 @@ const processUserInfo = (
...
     const accessToken = req.session.preloadedState?.session?.extraData?.oAuth2Data?.access_token;
     const refreshToken = req.session.preloadedState?.session?.extraData?.oAuth2Data?.refresh_token;
-    if(!accessToken || !refreshToken) {
+    const sessionExpiryTime = req.session?.preloadedState?.session_expires_at;
+    if(!accessToken || !refreshToken || !sessionExpiryTime) {
         return res.json({error: 'Access token or Refresh token not found'});
     }
+    // check if session set maxmum life is exceeded
+    if(new Date(Date.now()) >= new Date(sessionExpiryTime)) {
+        throw new Error(errorMessage);
```
Comment (moshthepitt) [inline src/app/index.ts:184]: Is the standard way to do this to throw an error?

**Review by KipSigei — APPROVED** [review]

File: src/app/index.ts [hunk 2]
```diff
@@ -168,12 +176,22 @@ const processUserInfo = (
...
+    // check if token refreshing is allowed
+    if(!EXPRESS_ALLOW_TOKEN_RENEWAL) {
+        return res.status(500).send({message: errorMessage});
+    }
     const accessToken = req.session.preloadedState?.session?.extraData?.oAuth2Data?.access_token;
     const refreshToken = req.session.preloadedState?.session?.extraData?.oAuth2Data?.refresh_token;
-    if(!accessToken || !refreshToken) {
+    const sessionExpiryTime = req.session?.preloadedState?.session_expires_at;
+    if(!accessToken || !refreshToken || !sessionExpiryTime) {
         return res.json({error: 'Access token or Refresh token not found'});
```
Comment (moshthepitt) [inline src/app/index.ts:188]: What kind of status code should this have?  Can we make it explicit?

Comment (moshthepitt) [inline src/app/index.ts:188]: Do we care a lot about magic strings here?

File: src/app/index.ts [hunk 3]
```diff
@@ -200,7 +200,7 @@ const refreshToken = (req: express.Request, res: express.Response) => {
             return res.json(preloadedState)
         })
         .catch((error: Error) => {
-            return res.json({error: error.message || 'Failed to refresh token'});
+            return res.status(500).send({message: error.message || TOKEN_REFRESH_FAILED});
```
Comment (ciremusyoka) [inline src/app/index.ts:203]: @moshthepitt am adding the status code on the response like in this line

**Review by moshthepitt — APPROVED** [review]

**Pull request merged**